jmix-flowui/flowui/src/main/java/io/jmix/flowui/sys/XmlInheritanceProcessor.java

```diff
@@ -0,0 +1,316 @@
+                    locator = l;
+                    break;
+                }
+            }
+            if (locator != null) {
+                Element target = locator.locate(resultElem, element);
+                // process target or a new element if target not found
+                if (target != null) {
+                    process(target, element);
+                } else {
+                    addNewElement(resultElem, element, justAdded);
+                }
+            } else {
+                // if no suitable locator found, look for a single element with the same name
+                List<Element> list = resultElem.elements(element.getName());
+                if (list.size() == 1 && !justAdded.contains(list.get(0))) {
+                    process(list.get(0), element);
+                } else {
+                    addNewElement(resultElem, element, justAdded);
+                }
+            }
+        }
+    }
+
+    protected void addNewElement(Element resultElem, Element element, Set<Element> justAdded) {
+        String idx = element.attributeValue(new QName("index", extNs));
+        Element newElement;
+        if (StringUtils.isBlank(idx)) {
+            newElement = resultElem.addElement(element.getName());
+        } else {
+            newElement = DocumentHelper.createElement(element.getName());
+
+            List<Element> elements = resultElem.elements();
+            int index = Integer.parseInt(idx);
+            if (index < 0 || index > elements.size()) {
+                String message = String.format(
+                        "Incorrect extension XML for screen. Could not paste new element %s to position %s",
+                        newElement.getName(), index);
+
+                throw new DevelopmentException(message,
+                        ParamsMap.of("element", newElement.getName(), "index", index));
+            }
+            elements.add(index, newElement);
+        }
+        justAdded.add(newElement);
+        process(newElement, element);
+    }
+
+    protected interface ElementTargetLocator {
+        boolean suitableFor(Element extElem);
+        @Nullable
+        Element locate(Element resultParentElem, Element extElem);
+    }
+
+    protected static class CommonElementTargetLocator implements ElementTargetLocator {
+
+        @Override
+        public boolean suitableFor(Element extElem) {
+            return !StringUtils.isBlank(extElem.attributeValue("id"));
+        }
+
+        @Nullable
+        @Override
+        public Element locate(Element resultParentElem, Element extElem) {
+            String id = extElem.attributeValue("id");
+            for (Element e : resultParentElem.elements()) {
+                if (id.equals(e.attributeValue("id"))) {
+                    return e;
+                }
+            }
+            return null;
+        }
+    }
+
+    protected static class ViewElementTargetLocator implements ElementTargetLocator {
```

Comment:
This `ElementTargetLocator` handles `view` elements which is `fetchPlan` in CUBA terms. 
1. Requires refactoring and testing if it can be applied to the `fetchPlan` element, considering inline fetch plans. 
2. Requires renaming as it clashes with `view` (aka screen) it self, `FeetchPlanElementTargetLocator`